src/group_theory/submonoid/operations.lean

```diff
@@ -93,13 +93,64 @@ def add_submonoid.of_submonoid {M : Type*} [add_zero_class M] (S : submonoid (mu
   zero_mem' := S.one_mem',
   add_mem' := S.mul_mem' }
 
+lemma submonoid.to_add_submonoid_coe {M : Type*} [mul_one_class M] (S : submonoid M) :
+  (S.to_add_submonoid : set (additive M)) = additive.of_mul '' S :=
+eq.symm (set.image_id S)
+
+lemma add_submonoid.to_submonoid_coe {M : Type*} [add_zero_class M] (S : add_submonoid M) :
+  (S.to_submonoid : set (multiplicative M)) = multiplicative.of_add '' S :=
```

Comment:
I think if you state this in terms of a preimage, then the proof is rfl

Reply:
Yes, `rfl` works using the preimage, but then in `submonoid.closure.to_add_submonoid` the proof become more annoying... at least, my proof is longer.

Reply:
Even if you change that lemma to also be about the preimage?

Reply:
Yes, in `submonoid.closure.to_add_submonoid` `S` is a subset, not a submonoid, and indeed I don't apply `add_submonoid.to_submonoid_coe` to `S`.